core/src/main/java/jenkins/scm/RunWithSCM.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2014 Jesse Glick.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.scm;
+
+import hudson.model.User;
+import hudson.scm.ChangeLogSet;
+
+import javax.annotation.Nonnull;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * @since 2.xxx
+ */
+public interface RunWithSCM {
+
+    @Nonnull List<ChangeLogSet<? extends ChangeLogSet.Entry>> getChangeSets();
+
+    @Nonnull Set<User> getCulprits();
```

Comment:
But fwiw, `getCulprits` actually will differ, since `AbstractBuild.getCulprits()` relies on `AbstractBuild.getDependencyChanges(...)`, which isn't functionality that I think makes sense in `WorkflowRun`. Especially since the `Fingerprinter` stuff is allllll about `AbstractProject`.